lib/publish-rsr.js

```diff
@@ -0,0 +1,96 @@
+import * as Sentry from '@sentry/node'
+import timers from 'node:timers/promises'
+import pRetry from 'p-retry'
+
+const ONE_HOUR = 60 * 60 * 1000
+
+const withPgClient = fn => async ({ createPgClient, ...args }) => {
+  const pgClient = await createPgClient()
+  try {
+    return await fn({ pgClient, ...args })
+  } finally {
+    await pgClient.end()
+  }
+}
+
+export const postEvaluate = withPgClient(async ({ pgClient, round, gitCommit }) => {
+  await pgClient.query(`
+    INSERT INTO publish_rsr_rounds
+    (round_index, spark_evaluate_version, measurement_commitments, round_details, providers)
+    VALUES
+    ($1, $2, $3, $4, $5)
+  `, [
+    round.index,
+    gitCommit,
+    round.cids,
+    round.details,
+    round.measurements.reduce((acc, m) => {
+      acc[m.minerId] = acc[m.minerId] || { successful: 0, total: 0 }
+      acc[m.minerId].total++
+      if (m.retrievalResult === 'OK') {
+        acc[m.minerId].successful++
+      }
+      return acc
+    }, {})
+  ])
+})
+
+const publishRsr = withPgClient(async ({ pgClient, storachaClient, rsrContract }) => {
+  const { rows } = await pgClient.query(`
+    SELECT *
+    FROM publish_rsr_rounds
+    WHERE evaluated_at <= now()::date AND evaluated_at > now()::date - interval '1 day'
```

Comment:
Is it safe to publish commitments for the current day, before the day is over? What if we end up adding one more evaluation after this code was executed?

I propose to select data from yesterday - that data is not going to change.

Reply:
`now()::date` is 0am today, so this is querying from yesterday's 0am to today's 0am, which is what you're suggesting. Right?

Reply:
> `now()::date` is 0am today, so this is querying from yesterday's 0am to today's 0am, which is what you're suggesting. Right?

I see! That's clever 👍🏻

Can we find a way how to make this condition easier to understand?

We want `evaluated_at` to be in the interval `now()::date - interval '1 day' to `now()::date`.

I would also include 0:00am in the same day, not in the previous day as you do know. It's unlikely this is going to make any difference in the result, but it could make the code easier to understand.

This is how I would write it myself:

```suggestion
    WHERE now()::date - interval '1 <= evaluated_at AND evaluated_at < now()::date 
```

Feel free to keep your current version if you don't see my proposal as a meaningful improvement.